climpred/bootstrap.py

```diff
@@ -51,6 +51,44 @@ def _resample(hind, resample_dim):
     return smp_hind
 
 
+def _resample_idx(init, bootstrap, dim='member', replace=True):
+    """Resample bootstrap times over dim."""
+
+    def xr_broadcast(x, idx):
+        return np.moveaxis(x.squeeze()[idx.squeeze().transpose()], 0, -1)
+
+    # I get errors if I have a 0 in dim
+    if 0 in init[dim].values:
+        init[dim] = np.arange(1, 1 + init[dim].size)
+    # resample with or without replacement
+    if replace:
+        idx = np.random.randint(0, init[dim].size, (bootstrap, init[dim].size))
+    elif not replace:
```

Comment:
I tested this against a more concise line:
```python
select_from = np.arange(init[dim].size)
np.vstack([np.random.choice(select_from, init[dim].size, replace=False) for _ in range(bootstrap)])
```

And yours is as much as 30x faster, so let's keep it this way.

Reply:
no change